hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -586,12 +596,28 @@ int MDMParser::waitFinalResp(_CALLBACKPTR cb /* = NULL*/,
                             else if (a == 4) *reg = REG_UNKNOWN;  // 4: unknown
                             else if (a == 5) *reg = REG_ROAMING;  // 5: registered, roaming
 
-                            // R410 : Look out for a CEREG: de-registration URC while attached
-                            if (_dev.dev == DEV_SARA_R410 && _attached && !REG_OK(*reg) && !strcmp(s, "CEREG:")) {
-                                MDM_PRINTF("Cell de-registered\r\n");
-                                _attached_urc = 0;
-                                _ip = NOIP;
+                            _reinitModem = 0xFF;
+                            if (mode == 0) {
+                                if (_dev.dev == DEV_SARA_R410) {
+                                    if (!strcmp(s, "CEREG:")) {
+                                        _reinitModem = 0x55;
+                                    }
+                                } else {
+                                    if (!strcmp(s, "CREG:") || !strcmp(s, "CGREG:")) {
+                                        _reinitModem = 0x55;
+                                    }
+                                }
+                            }
+                            if (_reinitModem == 0x55) {
+                                MDM_ERROR("[ Modem unexpectedly reset ]\r\n");
+                                _lastVerboseCxregUpdate = 0;
+                                // reinit handled in MDMParser::process()
+                            } else if (_dev.dev == DEV_SARA_R410 && _attached && !REG_OK(*reg) && !strcmp(s, "CEREG:")) {
+                                // R410 : Look out for a CEREG: de-registration URC while attached
+                                MDM_ERROR("[ Cell de-registered ]\r\n");
+                                _ip = NOIP; // Short-circuit disconnect() call
                                 _attached = false;
+                                _activated = false;
```

Comment:
This wasn't here originally. This change will break `reconnect()` call in `process()`.

Reply:
My bad 👍